deployments/gpu-operator/reconfigure-mig.go

```diff
@@ -75,40 +88,41 @@ func reconfigure(nodeName string, migConfigFile string, selectedMIGConfig string
 	*/
 	state, err := currentStateLabel(nodeName)
 	if err != nil {
-		setStateAndExit("failed", 1, currentLabels, hostGPUClientServices, withShutdownHostGPUClients, hostRootMount, nodeName)
+		setState("failed", 1, currentLabels, hostGPUClientServices, withShutdownHostGPUClients, hostRootMount, nodeName)
 		return err
 	}
 
 	/*
 		checking if selected MIG config is currently applied or not
-		i/p: migConfigFile, selectedMIGConfig
+		i/p: nvidiaMigPartedAlias, migConfigFile, selectedMIGConfig
 		o/p: True or false
 	*/
-	if err := migConfigCurrentlyApplied(nvidiaMigPartedAlias, migConfigFile, selectedMIGConfig); err == nil {
-		setStateAndExit("success", 0, currentLabels, hostGPUClientServices, withShutdownHostGPUClients, hostRootMount, nodeName)
+	if isMigConfigCurrentlyApplied(migConfigProperties, selectedMIGConfig) == true {
+		setState("success", 0, currentLabels, hostGPUClientServices, withShutdownHostGPUClients, hostRootMount, nodeName)
 		return nil
 	}
 
 	/*
-		host_persist_config
+		TODO: rightConfigApplied
 		ensuring that right config is applied to the node if the MIG Parted is also installed as a systemd service
-		i/p: hostRootMount, hostMIGManagerStateFile, selectedMIGConfig
+		i/p: hostRootMount, hostMIGManagerStateFile, selectedMIGConfig, currentLabels, hostGPUClientServices, nodeName, withShutdownHostGPUClients
 		o/p: true or false
+
+		if err := rightConfigApplied(hostRootMount, hostMIGManagerStateFile, selectedMIGConfig, currentLabels, hostGPUClientServices, nodeName, withShutdownHostGPUClients); err != nil{
```

Comment:
I think this function name is misleading. It does not properly reflect what we want to achieve here.